### ISSUE TITLE
BUG: Simplex+no presolve gives wrong model status

### ISSUE THREAD

**Issue by mckib2** [ERGO-Code/HiGHS#334]
Broken after pull request #327 (was working before this)

Based on [unit this test](https://github.com/scipy/scipy/blob/master/scipy/optimize/tests/test_linprog.py#L921) testing for infeasibility (MPS file attached at bottom).

Command run:
```
./highs --solver simplex --presolve off test_enzo_example_c_with_infeasibility.mps
```

Parameters:
- simplex
- no presolve

Should return infeasible status but gives unbounded:
```
LP       : test_enzo_example_c_with_infeasibility
Rows     : 2
Cols     : 50
Nonzeros : 98
INFO   : Solving LP without presolve or with basis
INFO   : Scaling: Improvement factor is nan >= 1 so scale LP
INFO   : Initial basis condition estimate of           1 is within the tolerance of 1e+14
INFO   : Using dual simplex solver - serial
       Iteration        Objective     Infeasibilities num(sum)
DuPh1          0    -1.5199588304e+02 Ph1: 2(78.0071); Du: 50(151.996)
DuPh1          2    -9.5164051867e-04 Ph1: 0(0); Du: 1(0.000951641)
DuPh1          2    -9.7656250000e-04 Ph1: 0(0); Du: 1(0.000976562)
INFO   : Not going to phase 2 immediately as hsol would, since dual objective is -0.0009766, not zero
INFO   : Phase 1 problem is optimal with original costs but num / max / sum dual infeasibilities = 1 / 0.0009766 / 0.0009766
INFO   : Dual simplex iterations [Ph1 2; Ph2 0; Pr 0] Total 2
INFO   : HiGHS basic solution: Analysis - Solving LP without presolve or with basis
INFO   : HiGHS basic solution: 2 Simplex iterations; Status: Unbounded
Postsolve  : 
Time       : 0.0012
For LP test_enzo_example_c_with_infeasibility: Solve original LP 0.00118 ( 99%)

Model   status      : Unbounded
Primal  status      : Not set
Dual    status      : Not set
Simplex   iterations: 2
HiGHS run time      :    1.1964e-03
```

IPM (presolve and no presolve) as well as simplex with presolve gives the expected infeasible status.

MPS file:
[test_enzo_example_c_with_infeasibility.txt](https://github.com/ERGO-Code/HiGHS/files/4549675/test_enzo_example_c_with_infeasibility.txt)

**Comment by mckib2** [comment]
i did a little digging and this might be because of the switch in how infinity is handled.  Notice in the log output that:
```
INFO   : Scaling: Improvement factor is nan >= 1 so scale LP
```

The `nan` is propagated through from `col_min_value` which has an initial value of `inf`.  However, when I make sure that `col_min_value` is not `inf` to begin with, the `nan` reappears at the calculation of `geomean_original_col`, so I may be incorrect in my diagnosis.  This could have been a "problem" all along and not related to the current issue.

**Comment by mckib2** [comment]
Pull request #338 fixes the nan issue, but the issue remains that the model status is incorrect.  Glad to have found the nan issue though!

**Comment by jajhall** [comment]
I see what I've done. It's not dual feasible, so I've deduced primal unboundedness. But that's only true if it's primal feasible. When an LP is both primal and dual infeasible then the status to report is primal infeasible. Now I see why Qi handled phase 1 optimality as he did! 

Sorry: I'll get this fixed today if I can.

The NaN issue was due to a trick of Qi's not working when HIGHS_CONST_INF is inf rather than 1e200!
